apiserver/common/tools.go

```diff
@@ -274,84 +286,69 @@ func (f *ToolsFinder) findTools(args params.FindToolsParams) (coretools.List, er
 	return fullList, nil
 }
 
-// findMatchingTools searches tools storage and simplestreams for tools matching the
-// given parameters. If an exact match is specified (number, series and arch)
-// and is found in tools storage, then simplestreams will not be searched.
-func (f *ToolsFinder) findMatchingTools(args params.FindToolsParams) (result coretools.List, _ error) {
-	exactMatch := args.Number != version.Zero && (args.OSType != "" || args.Series != "") && args.Arch != ""
+func (f *ToolsFinder) resultForOSTools(list coretools.List, osType string) coretools.List {
+	added := make(map[version.Binary]bool)
+	var matched coretools.List
+	for _, t := range list {
+		converted := *t
 
-	// TODO(juju4) - remove this logic
-	// Older versions of Juju publish agent binary metadata based on series.
-	// The following functions convert the result tools list to the required format.
-
-	// For a given list of tools, return those which match the required
-	// os type. If the tools is for a series, first convert the series into
-	// the corresponding os type before doing the match.
-	// eg if ubuntu tools are requested, and we have bionic tools, say they're ubuntu.
-	wantedOSType := args.OSType
-	compatibleOSTypeTools := func() {
-		added := make(map[version.Binary]bool)
-		var matched coretools.List
-		for _, t := range result {
-			converted := *t
-			// t might be for a series so convert to an os type.
-			if !coreos.IsValidOSTypeName(t.Version.Release) {
-				osTypeName, err := coreseries.GetOSFromSeries(t.Version.Release)
-				if err != nil {
-					continue
-				}
-				converted.Version.Release = strings.ToLower(osTypeName.String())
-			}
-			if converted.Version.Release != wantedOSType {
-				continue
-			}
-			if added[converted.Version] {
+		// t might be for a series so convert to an OS type.
+		if !coreos.IsValidOSTypeName(t.Version.Release) {
```

Comment:
I do always read that as https://getfedora.org/en/coreos?stream=stable vs core-os :)